crates/rome_js_parser/src/syntax/typescript/types.rs

```diff
@@ -226,12 +258,14 @@ pub(crate) fn parse_ts_type(p: &mut JsParser) -> ParsedSyntax {
                     let m = left.precede(p);
                     p.expect(T![extends]);
 
-                    p.with_state(EnterConditionalTypes::disallow(), parse_ts_type)
+                    parse_ts_type(p, context.and_conditional_type_allowed(false))
                         .or_add_diagnostic(p, expected_ts_type);
                     p.expect(T![?]);
-                    parse_ts_type(p).or_add_diagnostic(p, expected_ts_type);
+                    parse_ts_type(p, context.and_conditional_type_allowed(true))
+                        .or_add_diagnostic(p, expected_ts_type);
                     p.expect(T![:]);
-                    parse_ts_type(p).or_add_diagnostic(p, expected_ts_type);
+                    parse_ts_type(p, context.and_conditional_type_allowed(true))
+                        .or_add_diagnostic(p, expected_ts_type);
```

Comment:
Nit: `conditional_type_allowed` is guaranteed to be true because of the `is_conditonal_type_allowed` check above.